specfile/specfile.py

```diff
@@ -104,3 +115,68 @@ def tags(self, section: str = "package") -> Iterator[Tags]:
                 yield tags
             finally:
                 raw_section.data = tags.reassemble()
+
+    @contextlib.contextmanager
+    def changelog(self) -> Iterator[Optional[Changelog]]:
+        """
+        Context manager for accessing changelog.
+
+        Yields:
+            Spec file changelog as `Changelog` object or None if there is no %changelog section.
+        """
+        with self.sections() as sections:
+            try:
+                section = sections.changelog
+            except AttributeError:
+                yield None
+            else:
+                changelog = Changelog.parse(section)
+                try:
+                    yield changelog
+                finally:
+                    section.data = changelog.get_raw_section_data()
+
+    def add_changelog_entry(
+        self,
+        entry: Union[str, List[str]],
+        author: Optional[str] = None,
+        email: Optional[str] = None,
+        timestamp: Optional[Union[datetime.date, datetime.datetime]] = None,
+    ) -> None:
+        """
+        Adds a new %changelog entry. Does nothing if there is no %changelog section.
+
+        If not specified, author and e-mail will be determined using rpmdev-packager, if available.
+        Timestamp, if not set, will be set to current time (in local timezone).
+
+        Args:
+            entry: Entry text or list of entry lines.
+            author: Author of the entry.
+            email: E-mail of the author.
+            timestamp: Timestamp of the entry.
+              Supply `datetime` rather than `date` for extended format.
+        """
+        with self.changelog() as changelog:
+            if changelog is None:
+                return
+            evr = self.expand(
+                "%{?epoch:%{epoch}:}%{version}-%{release}", extra_macros=[("dist", "")]
+            )
+            if isinstance(entry, str):
+                entry = [entry]
+            if timestamp is None:
+                now = arrow.now()
+                # honor the timestamp format, but default to date-only
+                if changelog and changelog[-1].extended_timestamp:
+                    timestamp = now.datetime
+                else:
+                    timestamp = now.date()
+            if author is None:
+                try:
+                    author = subprocess.check_output("rpmdev-packager").decode().strip()
```

Comment:
This confused me a bit at first (setting email to None), maybe a comment mentioning that author from rpmdev-packager already contains the formatted mail could be useful.

Reply:
I'll do that. In fact, I'd prefer not separating e-mail and author's name and treat it as a single entity (RPM does so as well), but it was a requirement from packit.

Reply:
Good idea, I think that would be fine, the ChangelogEntry class will be simple and packit will have a high-level API for setting author and email separately. I was going to propose one more alternative approach (having an `Author` class that could be constructed either in the format of rpmdev-packager or from name and email) but that seems to be too complicated, especially from the caller's point of view so I'd go with what you proposed